lib/src/model/package_builder.dart

```diff
@@ -200,46 +139,87 @@ class PubPackageBuilder implements PackageBuilder {
   /// Parse a single library at [filePath] using the current analysis driver.
   /// If [filePath] is not a library, returns null.
   Future<DartDocResolvedLibrary> processLibrary(String filePath) async {
-    var name = filePath;
+//    var name = filePath;
+//
+//    if (name.startsWith(directoryCurrentPath)) {
+//      name = name.substring(directoryCurrentPath.length);
+//      if (name.startsWith(Platform.pathSeparator)) name = name.substring(1);
+//    }
+//    var javaFile = JavaFile(filePath).getAbsoluteFile();
+//    Source source = FileBasedSource(javaFile);
 
-    if (name.startsWith(directoryCurrentPath)) {
-      name = name.substring(directoryCurrentPath.length);
-      if (name.startsWith(Platform.pathSeparator)) name = name.substring(1);
-    }
     var javaFile = JavaFile(filePath).getAbsoluteFile();
-    Source source = FileBasedSource(javaFile);
-
-    // TODO(jcollins-g): remove the manual reversal using embedderSdk when we
-    // upgrade to analyzer-0.30 (where DartUriResolver implements
-    // restoreAbsolute)
-    var uri = embedderSdk?.fromFileUri(source.uri)?.uri;
-    if (uri != null) {
-      source = FileBasedSource(javaFile, uri);
-    } else {
-      uri = driver.sourceFactory.restoreUri(source);
-      if (uri != null) {
-        source = FileBasedSource(javaFile, uri);
-      }
-    }
-    var sourceKind = await driver.getSourceKind(filePath);
+    filePath = javaFile.getPath();
+
+    var analysisContext = contextCollection.contextFor(config.inputDir);
+    var session = analysisContext.currentSession;
+    var sourceKind = await session.getSourceKind(filePath);
+
     // Allow dart source files with inappropriate suffixes (#1897).  Those
     // do not show up as SourceKind.LIBRARY.
     if (sourceKind != SourceKind.PART) {
       // Loading libraryElements from part files works, but is painfully slow
       // and creates many duplicates.
-      final library =
-          await driver.currentSession.getResolvedLibrary(source.fullName);
+      print('[processLibrary][filePath: $filePath]');
```

Comment:
delete debug stuff?

Reply:
Yeah, of course I will not land it with all this debug output.
But the issue is that it fails on bots, but not locally.
So, I'm trying to debug it :-(
It is tough that I can only change `dartdoc`, while the issue might be somewhere in `analyzer`, and I really want to have debug output in there.

Reply:
Can you change dartdoc's dep on analyzer to be a path dependency, and add debug output at will?

Reply:
I can change the dependency on analyzer, and I can run tests locally.
They pass locally.
They fail on the bot.
I suspect there are two issues: 
1. with `.dart_tool/build`
2. something turns paths into local case on Windows.
I have had no time to dig deeper again.